tests/std/tests/P1135R6_barrier/test.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <barrier>
+#include <cassert>
+#include <chrono>
+#include <thread>
+
+void test() {
+    std::barrier barrier(2);
+
+    std::atomic<int> c{0};
+
+    std::thread t1([&] {
+        for (int i = 0; i < 5; i++) {
+            auto token = barrier.arrive();
+            barrier.wait(std::move(token));

```

Comment:
Technically should include `<utility>` for `std::move`. (I like to think that this will make a difference when we fully modularize.)